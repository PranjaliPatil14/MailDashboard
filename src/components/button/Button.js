import React from "react";
import * as PropTypes from "prop-types";
import "./button.css";

const Button = ({ children, onClick, className, testId }) => {
  return (
    <button
      type="submit"
      onClick={onClick}
      className={`button ${className}`}
      data-testid={testId}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  testId: PropTypes.string,
};

Button.defaultProps = {
  className: "",
  onClick: () => {},
  testId: "",
};

export default Button;
