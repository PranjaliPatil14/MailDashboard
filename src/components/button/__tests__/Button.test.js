import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../Button";

describe("Button", () => {
  const onClick = jest.fn();
  test("should render button", () => {
    const { asFragment, getByRole } = render(
      <Button onClick={onClick}>Text</Button>
    );

    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(getByRole("button"));

    expect(onClick).toHaveBeenCalled();
  });

  test("should render button when classname is passed", () => {
    const { asFragment } = render(
      <Button className="test-class">
        <span>span</span>
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
