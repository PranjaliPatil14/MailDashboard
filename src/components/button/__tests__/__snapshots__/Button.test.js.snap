// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render button 1`] = `
<DocumentFragment>
  <button
    class="button "
    data-testid=""
    type="submit"
  >
    Text
  </button>
</DocumentFragment>
`;

exports[`Button should render button when classname is passed 1`] = `
<DocumentFragment>
  <button
    class="button test-class"
    data-testid=""
    type="submit"
  >
    <span>
      span
    </span>
  </button>
</DocumentFragment>
`;
